src/dashboard/apps/core/factories.py

```diff
@@ -39,5 +41,9 @@ class DeliveryPointFactory(factory.django.DjangoModelFactory):
     class Meta:  # noqa: D106
         model = DeliveryPoint
 
-    provider_assigned_id = factory.Sequence(lambda n: "dp_%d" % n)
+    provider_assigned_id = factory.LazyAttributeSequence(
+        lambda obj, n: "".join(
+            str(random.randint(0, 9)) for _ in range(14)  # noqa: S311
+        )
+    )
```

Comment:
As you are not referring to the object (or `n`), you don't need a lazy attribute (sequence) ; a lazy function would be sufficient. Why not using a UUID instead?